### ISSUE TITLE
Load NeXus files containing event data directly into Scipp

### ISSUE THREAD

**Issue by matthew-d-jones** [scipp/scipp#1615]
Currently one has to load with Mantid and then convert the `EventWorkspace` to a Scipp `Variable`.

Use h5py for the first pass at this and we can assess later whether we think we can get better performance with C++.

Related to #1108

**Comment by SimonHeybrock** [comment]
Note that in principle `sc.bin` should be able to do much of what we need to sort events into pixels. I think it is currently lacking a mechanism to broadcast the pulse times, but otherwise it may be a decent starting point. Maybe broadcast pulse times by hand before calling `sc.bin`?

Performance may not be ideal since `sc.bin` is of course not optimized for this particular use-case.

**Comment by SimonHeybrock** [comment]
Based on https://github.com/scipp/scipp/blob/ad64e1151ffa5ff8bd83bfcb755683653c7e3e74/python/tests/test_bins.py#L61 we'd probably want to do:

```
events = events.bins.concatenate('pulse')
events = sc.bin(events, groups=[detector_id])
```

`bins.concatenate` does currently not broadcast the outer (bin) coords to the events. This is an open TODO and may be the way we should chose here for now, instead of making a dedicated implementation for sorting events?

**Comment by matthew-d-jones** [comment]
Do we want to sort as part of the load operation? I don't think `from_mantid()` sorts.
I guess it makes sense if every conceivable use case requires it.

**Comment by SimonHeybrock** [comment]
No need to sort by TOF, no. Scipp does not make as much use of sorting as Mantid does, since the generic data structures to not allow for maintaining such an internal state.

**Comment by matthew-d-jones** [comment]
First step towards what Simon describes above is recorded as issue #1627

**Comment by SimonHeybrock** [comment]
Fixed by scipp/scippneutron#17 and follow ups.
